glean-core/src/database/mod.rs

```diff
@@ -233,6 +233,14 @@ impl Database {
             writer.commit().unwrap();
         });
     }
+
+    /// Removes a metric from the storage (only Application Lifetime).
+    pub fn remove_application_lifetime_metric(&self, storage_name: &str, metric_name: &str) {
```

Comment:
Maybe it's better to have a generic `remove` method, to be symmetrical with `record`?  It seems weird that the caller would have to care about the lifetime of the metric.